Update zero react frontend lx (#21)
commitdev/zero-deployable-react-frontend: README.md

# Zero Deployable Frontend

This is a [Zero] module which sets up a

> service which can be deployed to the environment set up with [zero-aws-eks-stack][zero-infra].

Author: not sure about this suggestion, do you asking for the description to start from `line 4`?

> service which can be deployed to the environment set up with [zero-aws-eks-stack][zero-infra].

Reviewer: sorry I wrote something else must have messed it up somehow
```suggestion
React webapp bundled statically served through CloudFront and S3 from the [zero-aws-eks-stack][zero-infra].
```

The `/templates` folder is meant to be filled in via [Zero][zero] and results in Simple React application. It also contains a simple CircleCI pipeline which defines how to build and deploy the service.

> This repository is framework/business-logic agnostic; mainly showcasing some universal best practices:

Reviewer: ```suggestion
This repository is business-logic agnostic; mainly showcasing some universal best practices:
```

- Creating an optimized build of your frontend application.

- Out of the box CI/CD flow CircleCi.
  - testing

>   - uploading build to an AWS S3 bucket.

Reviewer: - building your web-app as a static distribution in /dist folder 
- syncs the built bundle to AWS s3
- invalidates Cloudfront cached version

## Repository structure

___

```sh
/   # file in the root directory is for initializing the user's repo and declaring metadata
|-- Makefile                        #make command triggers the initialization of repository
|-- zero-module.yml                 #module declares required parameters and 
|   # files in templates become the repo for users
|   templates/
|   |
|   |-- docs/

> |   |   | create-react-app.md       #contains documentation on how work with your react application and external links to developer resources.

Reviewer: ```suggestion
|   |   | create-react-app.md       #documentation on how to work with your react application and external developer resources.
```

|   |-- public/
|   |   |-- favicon.ico
|   |   |-- index.html
|   |   |-- logo192.png
|   |   |-- logo512.png
|   |   |-- manifest.json
|   |   |-- robots.txt
|   |-- src/                       #src contains your react application code
|   |-- .gitignore
|   |-- README.md
|   |-- package.json
|   |-- yarn.lock

```

## Prerequisites

- Have CircleCI and GitHub token setup with the Zero project.
- CI-user created via EKS-stack with access to AWS S3.

___

## Initialization

This step is meant to be executed during `zero apply`, includes following steps:

- Adding environment variables to CircleCI project.
- Linking the CircleCi with the GitHub repository

>   - Linking the circleCI will automatically trigger the first build and deploy your application to AWS S3

Reviewer: should probably talk about cloudfront and explain the wait time can be long
